### PR TITLE
feat: set cursor to 50% transparent when browser tab is in background

### PR DESCRIPTION
**Pull request by luzhenqian**

### CODE REVIEW

File: src/cursor/me.ts [hunk 1]
```diff
@@ -115,4 +117,27 @@ export default class Me extends Cursor {
...
+        const visibilityChange$ = fromEvent<MouseEvent>(
+            document,
+            'visibilitychange'
+        );
+
+        return visibilityChange$
+            .pipe(
+                map(() => {
+                    let event: 'leave' | 'enter';
+                    if (document.hidden) event = 'leave';
```
Comment (fanweixiao) [inline src/cursor/me.ts:131]: Would be better using [document.visibilityState](https://developer.mozilla.org/en-US/docs/Web/API/Document/visibilityState)

File: src/cursor/me.ts [hunk 2]
```diff
@@ -115,4 +117,27 @@ export default class Me extends Cursor {
...
+                    else event = 'enter';
+                    return {
+                        id: this.id,
```
Comment (fanweixiao) [inline src/cursor/me.ts:134]: seems `id` can be ignored here, cause this is must be `Me`

**Review by fanweixiao — APPROVED** [review]

File: src/OtherCursors.tsx [hunk 3]
```diff
@@ -0,0 +1,25 @@
+import React from 'react';
+import Others from './cursor/other';
+import OthersCursor from './OtherCursor';
+
```
Comment (xiaojian-hong) [inline src/OtherCursors.tsx:4]: In this PR, `Others` was renamed to `Other`, and `OthersCursor` to `OtherCursor`?

File: src/OtherCursor.tsx [hunk 4]
```diff
@@ -1,8 +1,9 @@
 import React, { useEffect, useMemo, useState } from 'react';
-import Others from './cursor/others';
+import Others from './cursor/other';
```
Comment (xiaojian-hong) [inline src/OtherCursor.tsx:2]: `Others` to `Other`

File: src/hooks/useOnlineCursor.ts [hunk 5]
```diff
@@ -57,7 +57,7 @@ const useOnlineCursor = ({
                             return old;
                         }
                         const cursorMap = new Map(old);
-                        const others = new Others(data);
+                        const others = new Other(data);
```
Comment (xiaojian-hong) [inline src/hooks/useOnlineCursor.ts:60]: `others` to `other`

File: src/OtherCursor.tsx [hunk 6]
```diff
@@ -1,8 +1,9 @@
...
 import CursorIcon from './CursorIcon';
 import Latency from './Latency';
 import useRenderPosition from './hooks/useRenderPosition';
+import useRenderOpacity from './hooks/useRenderOpacity';
 
 const OthersCursor = ({
```
Comment (xiaojian-hong) [inline src/OtherCursor.tsx:8]: The file `OthersCursor.tsx` was renamed to `OtherCursor.tsx`, so the class `OthersCursor` should be changed to `OtherCursor` as well in this file?

File: src/cursor/me.ts (hunk 1 above)
Comment (luzhenqian) [reply src/cursor/me.ts:131]: The effect of `document.hidden` and  `document.visibilityState === 'hidden'` are the same, Including performance and compatibility and so on.
But i think `document.hidden` is more direct and concise.
And many blog simple codes and source code will use `document.hidden` first.

Comment (luzhenqian) [reply src/cursor/me.ts:131]: > Would be better using [document.visibilityState](https://developer.mozilla.org/en-US/docs/Web/API/Document/visibilityState)

The effect of `document.hidden` and `document.visibilityState === 'hidden'` are the same, Including performance and compatibility and so on.
But i think `document.hidden` is more direct and concise, It is a high-level API suitable for use out of the box.
And many blog simple codes and source code will use `document.hidden` first.

File: src/cursor/me.ts (hunk 2 above)
Comment (luzhenqian) [reply src/cursor/me.ts:134]: > seems `id` can be ignored here, cause this is must be `Me`

it's right!

File: src/OtherCursors.tsx (hunk 3 above)
Comment (luzhenqian) [reply src/OtherCursors.tsx:4]: > In this PR, `Others` was renamed to `Other`, and `OthersCursor` to `OtherCursor`?

Yes, the cursor here means singular and should not be plural. Should only be plural when rendering with traverse.

File: src/hooks/useOnlineCursor.ts (hunk 5 above)
Comment (luzhenqian) [reply src/hooks/useOnlineCursor.ts:60]: > `others` to `other`

Got it.

File: src/OtherCursor.tsx (hunk 6 above)
Comment (luzhenqian) [reply src/OtherCursor.tsx:8]: I'll rename it.

Comment (luzhenqian) [reply src/OtherCursor.tsx:8]: > I'll rename it.

I'll rename it.

**Review by xiaojian-hong — APPROVED** [review]

**Review by fanweixiao — APPROVED** [review]

**Pull request merged**